mlflow/projects/utils.py

```diff
@@ -0,0 +1,127 @@
+    project_allowed_entries = ('name', 'conda_env', 'docker_env', 'entry_points')
+
+    project_entries = project_yaml.keys()
+
+    # make sure there are no extraneous entries
+    _validate_entries_are_allowed('project-level', project_entries, project_allowed_entries)
+
+    if 'name' in project_entries:
+        _validate_name_yaml(project_yaml.get('name'))
+
+    if 'docker_env' in project_entries:
+        _validate_docker_env_yaml(project_yaml.get('docker_env'))
+        # disallow conda and docker envs together in the same MLproject file
+        if 'conda_env' in project_entries:
+            multi_env_message = "cannot specify conda_env and docker_env in the same project"
+            raise ExecutionException(BAD_MLPROJECT_MESSAGE.format(multi_env_message))
+
+    if 'conda_env' in project_entries:
+        _validate_conda_env_yaml(project_yaml.get('conda_env'))
+
+    if 'entry_points' in project_entries:
+        for name, entry_point in project_yaml.get('entry_points').items():
+            _validate_entry_point_yaml(name, entry_point)
+
+
+def _validate_entries_are_allowed(yaml_level, present_entries, allowed_entries):
+    """helper to validate that entry keys are in an allowable set"""
+    for k in present_entries:
+        if k not in allowed_entries:
+            message = "{} entry {} not one of allowed entries {{{}}}"\
+                .format(yaml_level, k, ', '.join(allowed_entries))
+            raise ExecutionException(BAD_MLPROJECT_MESSAGE.format(message))
+
+
+def _validate_entry_point_yaml(entry_point_name, entry_point_yaml):
+    """helper to validate entry point yaml"""
+    entry_point_allowed_entries = ('parameters', 'command')
+
+    entry_point_entries = entry_point_yaml.keys()
+    _validate_entries_are_allowed('entry point', entry_point_entries, entry_point_allowed_entries)
+
+    if 'command' not in entry_point_entries:
+        no_command_message = "entry point {} has no corresponding command".format(entry_point_name)
+        raise ExecutionException(BAD_MLPROJECT_MESSAGE.format(no_command_message))
+
+    if 'parameters' in entry_point_entries:
+        for parameter_name, parameter_yaml in entry_point_yaml.get('parameters').items():
+            _validate_entry_point_parameter_yaml(entry_point_name, parameter_name, parameter_yaml)
+
+
+def _validate_entry_point_parameter_yaml(entry_point_name, parameter_name, parameter_yaml):
+    """helper to validate parameter yaml within an entry point"""
+    parameter_allowed_entries = ('type', 'default')
+
+    # define this here so we can provide a more informative message
+    def check_parameter_type(attempted_type):
+        if attempted_type not in MLPROJECT_PARAMETER_TYPES:
+            unsupported_parameter_type_message = (
+                "unsupported type for parameter {} in entry point {} will be converted to string"
+            ).format(parameter_name, entry_point_name)
+            print(unsupported_parameter_type_message)
```

Comment:
Maybe log this via `_logger.warn`, see an example of [initializing a logger for a module](https://github.com/mlflow/mlflow/blob/ed6fa273e2188ffc5c4673b03857fa1469411598/mlflow/projects/__init__.py#L53) & [using it](https://github.com/mlflow/mlflow/blob/ed6fa273e2188ffc5c4673b03857fa1469411598/mlflow/projects/__init__.py#L262)